hazelcast/src/main/java/com/hazelcast/instance/impl/ClusterTopologyIntent.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance.impl;
+
+public enum ClusterTopologyIntent {
+    /**
+     * Hazelcast is not deployed in a managed context
+     */
+    NOT_IN_MANAGED_CONTEXT,
+    /**
+     * Unknown state, but in managed context (Kubernetes)
+     */
+    UNKNOWN,
+    /**
+     * No change to the number of Hazelcast members in the cluster is intended
+     */
+    STABLE,
```

Comment:
are all other state either than `NOT_IN_MANAGED_CONTEXT` in managed context? A javaDoc for this also will be helpful for future reference.

Reply:
yes, explained in class javadoc added in https://github.com/hazelcast/hazelcast/pull/21844/commits/d0997b4f066efde9c3686094e114258c079a1702